### PR TITLE
Dividing by zero, display error and "ERROR" = 0

### PR DESCRIPTION
**Pull request by Smuger**
Showing ERROR when dividing by 0

### CODE REVIEW

File: src/components/App/App.tsx [hunk 1]
```diff
@@ -34,18 +34,33 @@ export const App: FunctionComponent = () => {
...
         newResult /= rightOperand
+        break
     }
 
     setResult(newResult)
-    setDisplay(newResult.toString().toString().slice(0, 12))
-
+    
+    // Display "ERROR" when /0 
+    if (pendingOperator === "÷" && rightOperand === 0){
```
Comment (tuanpham-dev) [inline src/components/App/App.tsx:47]: Need a space before open curly bracket

```suggestion
    if (pendingOperator === "÷" && rightOperand === 0) {
```

File: src/components/App/App.tsx [hunk 2]
```diff
@@ -34,18 +34,33 @@ export const App: FunctionComponent = () => {
...
+      setDisplay("ERROR")
+    }
+    else{
```
Comment (tuanpham-dev) [inline src/components/App/App.tsx:50]: Need to merge 2 lines into 1 and add space after close curly bracket and add space before open curly bracket

```suggestion
    } else {
```

File: src/components/App/App.tsx [hunk 3]
```diff
@@ -34,18 +34,33 @@ export const App: FunctionComponent = () => {
...
+      setDisplay(newResult.toString().toString().slice(0, 12))
+    }
     return true
   }
 
+  // Handling display="ERROR"
+  const convertToNumberWithCheckIfNaN = (display: string) => {
+    if (Number.isNaN(Number(display))){
```
Comment (tuanpham-dev) [inline src/components/App/App.tsx:58]: Need a space before open curly bracket

```suggestion
    if (Number.isNaN(Number(display))) {
```

File: src/components/App/App.tsx [hunk 4]
```diff
@@ -34,18 +34,33 @@ export const App: FunctionComponent = () => {
...
-
+    
+    // Display "ERROR" when /0 
+    if (pendingOperator === "÷" && rightOperand === 0){
+      setDisplay("ERROR")
+    }
+    else{
+      setDisplay(newResult.toString().toString().slice(0, 12))
+    }
     return true
```
Comment (tuanpham-dev) [inline src/components/App/App.tsx:53]: Need a newline before return statement

```suggestion

    return true
```

File: src/components/App/App.tsx [hunk 5]
```diff
@@ -34,18 +34,33 @@ export const App: FunctionComponent = () => {
...
   }
 
+  // Handling display="ERROR"
+  const convertToNumberWithCheckIfNaN = (display: string) => {
+    if (Number.isNaN(Number(display))){
+      display = "0"
+    }
+    return Number(display)
```
Comment (tuanpham-dev) [inline src/components/App/App.tsx:61]: Need a newline before return statement

```suggestion

    return Number(display)
```

File: src/components/App/App.tsx [hunk 6]
```diff
@@ -34,18 +34,33 @@ export const App: FunctionComponent = () => {
...
+  }
```
Comment (tuanpham-dev) [inline src/components/App/App.tsx:62]: Assign Number(display) to a variable

```suggestion
  const convertToNumberWithCheckIfNaN = (display: string): number => {
    const value = Number(display)
    
    if (Number.isNaN(value)) {
      return 0
    }
    
    return value
  }
```

**Comment by Smuger** [comment]
Thanks

**Pull request closed without merging**